src/lib.rs

```diff
@@ -310,4 +407,33 @@ mod tests {
             new_gitbom.get_sorted_oids()[0].hex_hash()
         )
     }
+
+    #[tokio::test]
+    async fn test_async_read() {
+        let mut to_read = Vec::new();
+        for _ in 0..50 {
+            to_read.push((
+                tokio::fs::File::open("test/data/hello_world.txt")
+                    .await
+                    .unwrap(),
+                11,
```

Comment:
Okay... this one also had me scratching my head...

We need to know the length of the stream at the beginning of the `Hash.digest` phase as this is part of how Git generates hashes.

Using some form of abstract Reader thingy decouples the library from being file oriented. I always like abstractions because without them, the concrete "it's a file on the filesystem" or whatever the other part of the concreteness oozes through and permeates the code.

So... I'm going to reject your recommendation at this point. As we start adding some CLI stuff to the library/executable, we may find a helper function like the one you're suggesting makes sense.

However, I will adopt the `Source` rather than tuple recommendation above.